grakn-engine/src/test/java/ai/grakn/engine/task/postprocessing/PostProcessingTaskTest.java

```diff
@@ -94,4 +94,34 @@ public void whenTheIndexCacheIsEmpty_NothingStarts(){
         //Check no methods calls
         verify(indexPostProcessor, Mockito.times(0)).mergeDuplicateConcepts(any(), any(), any());
     }
+
+    @Test
+    public void whenThereAreMultipleIndicesToPostProcess_EnsurePPIsCalledMultipleTimes() throws InterruptedException {
+        String index1 = "index1";
+        String index2 = "index2";
+        String index3 = "index3";
+        String index4 = "index4";
+
+        when(indexPostProcessor.popIndex(keyspaceA)).
+                thenReturn(index1).
+                thenReturn(index2).
+                thenReturn(index3).
+                thenReturn(index4).
+                thenReturn(null);
+
+        Set<ConceptId> ids = Stream.of("id1", "id2", "id3").map(ConceptId::of).collect(Collectors.toSet());
+        when(indexPostProcessor.popIds(keyspaceA, index1)).thenReturn(ids);
+        when(indexPostProcessor.popIds(keyspaceA, index2)).thenReturn(ids);
+        when(indexPostProcessor.popIds(keyspaceA, index3)).thenReturn(ids);
+        when(indexPostProcessor.popIds(keyspaceA, index4)).thenReturn(ids);
+
+        //Run the method
+        postProcessingTask.run();
+
+        //Give time for PP to run
+        Thread.sleep(config.getProperty(GraknConfigKey.POST_PROCESSOR_DELAY) * 2000);
```

Comment:
Why 2000?

Reply:
i think this is the part where we wait for cassandra index to be ready?

Reply:
I'm just wondering about the exact number - sleeping in tests is a bit dangerous, so we should make sure we're sleeping for the right length.

Reply:
Ya 2 seconds is just to give time for the index to stabilise. I agree that sleeping is dangerous but we have no other way to check the status of the index right now.